definition_files.rst

```diff
@@ -771,6 +749,344 @@ After building the help can be displayed like so:
+.. code::
+
+   $ {command} build -F --build-arg file_contents="'I am in a file (1st time)'" --build-arg var_value2="'I am also in an env var'" --build-arg file_contents="'I am in a file (2nd time)'" --build-arg-file ./my_args_file.txt ./my_container.sif ./my_container.def
+   INFO:    Starting build...
+   Getting image source signatures
+   Copying blob f4bb4e8dca02 skipped: already exists  
+   Copying config 2b7cc08dcd done   | 
+   Writing manifest to image destination
+   2024/03/08 14:42:15  info unpack layer: sha256:f4bb4e8dca02be491b4f72d2ef2127a64ce49c48d0d9c0a0fcaffa625067679d
+   INFO:    Running setup scriptlet
+   + echo 'I am in a file (2nd time)'
+   INFO:    Adding environment to container
+   INFO:    Adding runscript
+   INFO:    Creating SIF file...
+   INFO:    Build complete: ./my_container.sif
+
+   $ {command} run ./my_container.sif
+   file contents: I am in a file (2nd time)
+   --> this should, if I'm not mistaken, equal: I am in a file (2nd time)
+
+   env var: I am in an env var
+   --> this should, if I'm not mistaken, equal: I am in an env var
+
+   env var: I am also in an env var
+   --> this should, if I'm not mistaken, equal: I am also in an env var
+
+   and finally, here's some text: I am just some text
+
+In the event that a variable is defined both in the file passed to
+``--build-arg-file`` and via the command line using ``--build-arg`` flag, the value passed via the
+command line will take precedence:
+
+.. code::
+
+   $ cat << EOF > ./my_args_file.txt
+   var_value1="I am in an env var"
+   var_value2="I am also in an env var (from build arg file)"
+   some_text="I am just some text"
+   EOF
+
+   $ {command} build -F --build-arg file_contents="'I am in a file'" --build-arg var_value2="'I am also in an env var (from command line)'" --build-arg-file ./my_args_file.txt ./my_container.sif ./my_container.def
+   INFO:    Starting build...
+   Getting image source signatures
+   Copying blob f4bb4e8dca02 skipped: already exists  
+   Copying config 2b7cc08dcd done   | 
+   Writing manifest to image destination
+   2024/03/08 14:42:15  info unpack layer: sha256:f4bb4e8dca02be491b4f72d2ef2127a64ce49c48d0d9c0a0fcaffa625067679d
+   INFO:    Running setup scriptlet
+   + echo 'I am in a file'
+   INFO:    Adding environment to container
+   INFO:    Adding runscript
+   INFO:    Creating SIF file...
+   INFO:    Build complete: ./my_container.sif
+
+   $ {command} run ./my_container.sif
+   file contents: I am in a file
+   --> this should, if I'm not mistaken, equal: I am in a file
+
+   env var: I am in an env var
+   --> this should, if I'm not mistaken, equal: I am in an env var
+
+   env var: I am also in an env var (from command line)
+   --> this should, if I'm not mistaken, equal: I am also in an env var (from command line)
+
+   and finally, here's some text: I am just some text
+
+Default variable values: the %arguments section
+===============================================
+
+.. _template arguments:
```

Comment:
I don't see any other examples of references with spaces in them; please use an underscore instead.